File: Basic Programming/Hackerrank 30 days of code python3 solution/Day_25.py
#                      Name : Arif Khan
#                      Judge: HACKERRANK
#                      University: Primeasia University
#                      problem: Day 25: Running Time and Complexity
#                      Difficulty: Easy
#                      Problem Link: https://www.hackerrank.com/challenges/30-running-time-and-complexity/problem
#

import math

def primecheck(n):
    x = int(math.sqrt(n))
    for i in range(2,x+1):
        if n%i == 0:
            return False
    return True

n = int(input())
while n:
    x = int(input())
    print('Prime' if primecheck(x) and x >= 2 else 'Not prime')
    n -= 1

File: Basic Programming/Hackerrank 30 days of code c++ solution/Day_25.cpp
/*                    Name : Arif Khan
                      Judge: HACKERRANK
                      University: Primeasia University
                      problem: Day 25: Running Time and Complexity
                      Difficulty: Easy
                      Problem Link: https://www.hackerrank.com/challenges/30-running-time-and-complexity/problem
*/

#include <cmath>
#include <cstdio>
#include <vector>
#include <iostream>
#include <algorithm>
using namespace std;

bool primech(int n){
    for(int i = 2; i<= sqrt(n) ;i++){
        if(n%i == 0)
            return false;
    }
    return true;
}

int main() {
    int n;
    cin>>n;
    while(n--){
        long t;
        cin >> t;
        if(primech(t) and t>=2) cout<<"Prime"<<endl;
            else cout<<"Not prime"<<endl;
    }
    return 0;
}